packages/core/src/odata/v4/payload-value-converter.ts

```diff
@@ -40,117 +43,74 @@ type EdmTypeMapping = {
-  if (momentInstance['_isUTC']) {
-    const offset = Math.abs(momentInstance.utcOffset());
-    const operator = momentInstance.utcOffset() >= 0 ? '+' : '-';
-    return `/Date(${timestamp}${operator}${leftpad(offset, 4)})/`;
+function edmDateTimeOffsetToMoment(dateTime: string): Moment {
+  const prefix = 'YYYY-MM-DDTHH:mm';
+  // In moment the Z is either Offset from UTC as +-HH:mm, +-HHmm, or Z
+  const validFormats = [`${prefix}Z`, `${prefix}:ssZ`, `${prefix}:ss.SSSZ`];
+  const parsed = moment(dateTime, validFormats, true);
+  if (!parsed.isValid()) {
+    throw new Error(
+      `Provided date-time value ${dateTime} does not follow the Edm.DateTimeOffset pattern: YYYY-MM-DDTHH:mm(:ss(.SSS))Z`
+    );
   }
-
-  return `/Date(${timestamp})/`;
+  return parsed;
 }
 
-/**
- * @hidden
- */
-export function parseNumber(value: string | number): number {
-  if (typeof value === 'number') {
-    return value;
+function edmDurationToMoment(value: string): Duration {
+  const durationPattern = /([\+,\-]{1,1})?P(\d+D)?T(\d{1,2}H)?(\d{1,2}M)?(\d{1,2}S)?(\d{2,2}\.\d+S)?/;
+  const captured = durationPattern.exec(value);
+  if (!captured || captured[0] !== value) {
+    throw new Error(
+      `Provided duration value ${value} does not follow the Edm.Duration pattern: +/- P0DT0H0M0S`
+    );
   }
+  return moment.duration(value);
+}
 
-  if (value.toLowerCase() === 'inf') {
-    return Number.POSITIVE_INFINITY;
-  }
-  if (value.toLowerCase() === '-inf') {
-    return Number.NEGATIVE_INFINITY;
-  }
-  if (value.toLowerCase() === 'nan') {
-    return Number.NaN;
+function edmTimeOfDay(value: string): Moment {
+  const prefix = 'HH:mm:ss';
+  const validFormats = [`${prefix}`, `${prefix}.SSS`];
+  const parsed = moment.utc(value, validFormats, true);
+  if (!parsed.isValid()) {
+    throw new Error(
+      `Provided time-of-day value ${value} does not follow the Edm.TimeOfDay pattern: HH:mm:ss(.SSS)`
+    );
   }
+  return moment.utc({
+    y: 1970,
+    M: 0,
+    d: 0,
+    h: parsed.hours(),
+    m: parsed.minutes(),
+    s: parsed.seconds(),
+    ms: parsed.milliseconds()
+  });
+}
 
-  const num = Number(value);
+function momentToEdmDate(value: Moment): string {
+  return value.format('YYYY-MM-DD');
+}
 
-  if (Number.isNaN(num)) {
-    throw new Error(`Cannot create number from input "${value}"`);
-  }
+function momentToEdmDateTimeOffsetToMoment(value: Moment): string {
+  return value.utc().format('YYYY-MM-DDTHH:mm:ss.SSS') + 'Z';
+}
 
-  return num;
+function durationToEdmDuration(value: Duration): string {
+  return value.toISOString();
 }
 
-function leftpad(value: any, targetLength: number): string {
-  const str = value.toString();
-  if (str.length >= targetLength) {
-    return str;
-  }
-  return '0'.repeat(targetLength - str.length) + str;
+function momentToEdmTimeOfDay(value: Moment): string {
+  return value.format('HH:mm:ss.SSS');
 }
 
 // Prettier-ignore
```

Comment:
Also not part of this PR, but can this be removed?

Reply:
removed it.